docs/Partitioner.md

```diff
@@ -0,0 +1,52 @@
+## Graph Partitioning Support in Glow
+
+### Overview
+Along with the increased size of neural networks, a large network may not be able to run on only one accelerator/device due to resource limitation (e.g. memory capacity). Therefore, it is necessary to divide up the network into several sub-networks and each of them can be executed on one accelerator. This problem can be abstracted as a Graph Partition problem [https://en.wikipedia.org/wiki/Graph_partition] : Given a neural network and n devices, we would like to split this network into minimal k sub-networks (where k <= n) , and for each sub-network, it can be run on one device with the device resource restriction.
+
+### Heterogeneous partition
+Glow now supports Heterogeneous Partition to address this problem. Given a network, and the information of a list of devices proviced by users, Glow will try to find a posssible partition or report an error.
+
+#### Design
+The figure below shows the Heterogeneous Partition flow: a network(i.e. Glow function) is first divided up into sub-networks based on backend types and then each sub-network is devided up based on given cost model. 
+![](HeterogeneousPartition.png)
+
+For Backend-Based-Partition : If a node/operator is only supported by backend A, it will be assigned to A. If a node is supported by both backends A and B, Glow choses the one with higher priority. In addition, users can define backend-support/backend-unsupport op list. This feature allows users to run one network cross different type of backends. E.g. users can choose any node running on their accelorator and the rest running on CPU.
+
+#### Usage
+[```struct DeviceInfo```](https://github.com/pytorch/glow/blob/master/include/glow/Runtime/RuntimeTypes.h#L50) is required to run Partitioner. [Here](https://github.com/pytorch/glow/blob/master/tests/unittests/PartitionerTest.cpp#L614) is an exmpale of how to call heterogeneous partition. 
```

Comment:
pinning to master and specific line is error prone.

Reply:
Could you please try it in "view file" mode?  In that mode, the link works for me.

Reply:
it only works with current master, if there will be changes in this file then the link will point to the wrong location.
You can use specific sha and not master and point to the line number.

Reply:
I think using master would keep the example up to date. I will sync the doc if the file changes :)

Reply:
ok, docs might be stale in either case.